libraries/AP_Baro/AP_Baro_ICM20789.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "AP_Baro_Backend.h"
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/Semaphores.h>
+#include <AP_HAL/Device.h>
+
+#ifndef HAL_BARO_ICM20789_I2C_ADDR
+// the baro is on a separate I2C address from the IMU, even though in
+// the same package
+#define HAL_BARO_ICM20789_I2C_ADDR 0x63
+#endif
+
+class AP_Baro_ICM20789 : public AP_Baro_Backend
+{
+public:
+    void update();
+
+    static AP_Baro_Backend *probe(AP_Baro &baro, AP_HAL::OwnPtr<AP_HAL::Device> dev);
+    
+private:
+    AP_Baro_ICM20789(AP_Baro &baro, AP_HAL::OwnPtr<AP_HAL::Device> dev);
+    virtual ~AP_Baro_ICM20789(void) {};
```

Comment:
Any reason to define the destructor?

Reply:
Probably because "that's how all the backends do it".

Will remove all the empty backend destructors.